src/main/kotlin/de/smartsquare/starter/mqtt/MqttPublisher.kt

```diff
@@ -50,4 +51,20 @@ class Mqtt5Publisher(private val adapter: MqttMessageAdapter, client: Mqtt5Clien
                 .build(),
         )
     }
+
+    /**
+     * Publishes the given [payload] on [topic] with quality of service level [qos] expiring with [expiry]. Expiry is
+     * rounded to full seconds.
+     * Returns a [CompletableFuture] that is completed once the broker has accepted the message.
+     */
+    fun publish(topic: String, qos: MqttQos, payload: Any, expiry: Duration): CompletableFuture<Mqtt5PublishResult> {
```

Comment:
The `Mqtt5PublishBuilder` supports many more options like `contentType()` or `responseTopic()`. I am a bit hesistant to add a single overload like since this we will end up with many permutations if we want to add other options later.

How about adding a new data class with all additional options that can be set optionally? Something like this:

```kt
/**
 * Publishes the given [payload] on [topic] with quality of service level [qos].
 * [options] can be passed to further customize.
 * Returns a [CompletableFuture] that is completed once the broker has accepted the message.
 */
@JvmOverloads
fun publish(
    topic: String,
    qos: MqttQos,
    payload: Any,
    options: Mqtt5PublishOptions? = null,
): CompletableFuture<Mqtt5PublishResult> {
    return asyncClient.publish(
        Mqtt5Publish.builder()
            .topic(topic)
            .qos(qos)
            .payload(adapter.adapt(payload))
            .apply { if (options?.expiry != null) messageExpiryInterval(options.expiry.toSeconds()) }
            .build(),
    )
}

data class Mqtt5PublishOptions(
    val expiry: Duration?,
    val contentType: String?,
)
```

`@JvmOverloads` allows us to keep being backwards compatible with java I think. We should add all available options and also have the same for mqtt 3 if we decide to go this route.

Reply:
This is a good idea. I did implement it for MQTT 5, however, for MQTT 3, there is only one option, which is `retain`. I don't think it makes sense to create a wrapper object just for this, so instead I just added that flag